### PR TITLE
First "ready for review" draft of the 2022 conda survey results.

### PR DESCRIPTION
**Pull request by tnabtaf**
Hi All,

This is a lengthy blog post about the conda survey that Anaconda ran towards the end of 2022.

I would very much appreciate any 

* editorial / content input
* formatting input, especially with how the images are handled,
* input on how the images are stored

I am looking for co-authors too, so if you update it enough, then please add yourself.

Thanks,
Dave C

### CODE REVIEW

File: blog/2023-03-10-conda-survey.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+* 71% prioritized speedups
+* 60% use mamba, which is known for being very fast compared to the standard conda solver.
+
+Some individual responses:
+
+* "just make libmamba the default solver already!!!!"
+* "Nearly 100% use Mamba nowadays due to speed, only fallback to Conda when something is going wrong"
+* "*Dear Goddess make the solver faster out of the box.*"
+
+This survey was taken in November and December 2022, which we feel was an inflection point in conda's performance profile.  The [22.11.0 conda release], which came out towards the end of the survey window, implemented parallel package download and extraction, and dropped the `experimental` tag from the `conda-libmamba-solver` (the solver was added in the March 2022 release). At the time of the survey, [only 19% of responsdents were using conda-libmamba-solver](https://docs.google.com/spreadsheets/d/1rpWN6atSEsALvStPtAVIf_WMTxXXKaDKYpKRsvMpbOo/edit#gid=1650032116).
```
Comment (jaimergp) [inline blog/2023-03-10-conda-survey.md:60]:
```suggestion
This survey was taken in November and December 2022, which we feel was an inflection point in conda's performance profile.  The [22.11.0 conda release], which came out towards the end of the survey window, implemented parallel package download and extraction, and dropped the `experimental` tag from the `conda-libmamba-solver` (the solver was added in the March 2022 release). At the time of the survey, [only 19% of respondents were using conda-libmamba-solver](https://docs.google.com/spreadsheets/d/1rpWN6atSEsALvStPtAVIf_WMTxXXKaDKYpKRsvMpbOo/edit#gid=1650032116).
```

Comment (jaimergp) [reply blog/2023-03-10-conda-survey.md:60]: Are we missing a [22.11.0 conda release] link?

File: blog/2023-03-10-conda-survey.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+Data Usage & ETL (67%, Extract, Transform and Load) and Machine Learning & AI (63%) were the most common applications.  Visualization (54%), Packaging (54%) and Scientific (53%) make up the next largest groups.
+
+# What is conda used with?
+
+## Packages
+
+The conda survey asked people to list their top 5 packages.
+
+[![Top 5 packages](/img/2023-03-10-conda-survey/top-5-packages.png)](https://docs.google.com/spreadsheets/d/1rpWN6atSEsALvStPtAVIf_WMTxXXKaDKYpKRsvMpbOo/edit#gid=968125862)
```
Comment (jaimergp) [inline blog/2023-03-10-conda-survey.md:134]: This image 404s on the preview.

File: blog/2023-03-10-conda-survey.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+Several other questions revealed a strong preference for just one tool/workflow for package management.
+
+[![How much do you agree with these statements?](/img/2023-03-10-conda-survey/psf-how-much-do-you-agree-with-these-statements.png)](/img/2023-03-10-conda-survey/psf-how-much-do-you-agree-with-these-statements.png)
+
+The first question is "I prefer to have a clearly defined "official" workflow" which 75% agreed with.  A second similar (but inversely worded) question asks if "The existence of multiple tools is beneficial for the Python packaging ecosystem.  Here 46% disagree with the statement, and 23% are neutral
+
+**What do we do with all this?**
+
+Let's start by noting that the Python community strongly prioritizes interoperability between Python packaging tool, but that support for multiple languages is at the bottom.  This is good news and bad news for conda.  Conda already works well with Pip.  The downside is that most PSF respondents don't prioritize support for multiple languages, which is one of conda's superpowers.
```
Comment (jaimergp) [inline blog/2023-03-10-conda-survey.md:106]: Consistency in `Pip` capitalization. Previous occurrences used `pip`. 

[`pypa/pip`](https://github.com/pypa/pip) uses `pip` (lowercase), always, even at the beginning of sentences.

`venv` should also go in lowercase, and maybe even backticked. [See docs](https://docs.python.org/3/library/venv.html).

Comment (jaimergp) [reply blog/2023-03-10-conda-survey.md:106]: (Same applies in the paragraph below)

File: blog/2023-03-10-conda-survey.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+[![Top 5 packages](/img/2023-03-10-conda-survey/top-5-packages.png)](https://docs.google.com/spreadsheets/d/1rpWN6atSEsALvStPtAVIf_WMTxXXKaDKYpKRsvMpbOo/edit#gid=968125862)
+
+This question had a few clear leaders with Numpy and Pandas used by over half of respondents.  Matplotlib, Scipy, and Scikit-learn were used by 1/4 to 1/5 of respondents.
+
+## Compute Environments
+
+Linux was the [dominant operating system](https://docs.google.com/spreadsheets/d/1rpWN6atSEsALvStPtAVIf_WMTxXXKaDKYpKRsvMpbOo/edit#gid=1402353521) with almost 70% use, but macOS and Windows weren't that far behind, both in the neighborhood of 50%. VS Code was the clearly [preferred IDE](https://docs.google.com/spreadsheets/d/1rpWN6atSEsALvStPtAVIf_WMTxXXKaDKYpKRsvMpbOo/edit#gid=1423420551) with 70% use.  The Jupyter platforms came in next, with both in the mid 40 percents.  PyCharm was used by 1/3 of paricipants.
+
+[![Programming languages, top](/img/2023-03-10-conda-survey/programming-languages-top.png)]
+(https://docs.google.com/spreadsheets/d/1rpWN6atSEsALvStPtAVIf_WMTxXXKaDKYpKRsvMpbOo/edit#gid=681062781)
```
Comment (jaimergp) [inline blog/2023-03-10-conda-survey.md:143]:
```suggestion
[![Programming languages, top](/img/2023-03-10-conda-survey/programming-languages-top.png)](https://docs.google.com/spreadsheets/d/1rpWN6atSEsALvStPtAVIf_WMTxXXKaDKYpKRsvMpbOo/edit#gid=681062781)
```

Comment (jaimergp) [reply blog/2023-03-10-conda-survey.md:143]: (single line required for link)

File: blog/2023-03-10-conda-survey.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+The most common role was *Developer*, followed closely by *Research Scientists.* The top 12 roles can be roughly grouped:
+
+* *Developer* (29.2%)
+* *Research Scientist* (27.8%) and *Applied Scientist* (15.3%)
+* *Data Scientist* (22.2%), *Data Engineer* (15.3%), and *ML Scientist* (12.5%)
+* *Professor/Instructor* (12.5%) and *Student* (11.1%)
+* *ML Ops* (6.9%), *DevOps* (5.6%), and *System Administrator* (4.2%)
+* *Project Manager* (4.2%)
+
+People could select more than one job function.
```
Comment (jaimergp) [inline blog/2023-03-10-conda-survey.md:194]: Oh, it ended already :(

I was expecting something like a conclusion or a closing statement or something like that. The end was a bit abrupt and could use a goodbye :D

File: blog/2023-03-10-conda-survey.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+The need for speed was the most prominent theme:
+
+* 71% prioritized speedups
+* 60% use mamba, which is known for being very fast compared to the standard conda solver.
+
+Some individual responses:
+
+* "just make libmamba the default solver already!!!!"
+* "Nearly 100% use Mamba nowadays due to speed, only fallback to Conda when something is going wrong"
+* "*Dear Goddess make the solver faster out of the box.*"
```
Comment (jaimergp) [inline blog/2023-03-10-conda-survey.md:58]: Check that bullet items all end in a period (or none, but having a mix is inconsistent).

File: blog/2023-03-10-conda-survey.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+## Better interoperability
+
+*Improved interoperability with other package managers* was prioritized by 62% of respondents. This message is reinforced by how frequently people use more than one package manager.
+
+[![Which additional package managers do your use, top](/img/2023-03-10-conda-survey/which-additional-package-managers-do-you-use-top.png)](https://docs.google.com/spreadsheets/d/1rpWN6atSEsALvStPtAVIf_WMTxXXKaDKYpKRsvMpbOo/edit#gid=1102296811)
+
+79% of conda survey respondents use other package managers in addition to conda.  75% of all respondents also use pip, and 55% use mamba.
+
+The PSF survey asked "What Packaging tools do you use?"  They blended together package and environment management (as does conda itself):
```
Comment (jaimergp) [inline blog/2023-03-10-conda-survey.md:83]:
```suggestion
The PSF survey asked "What Packaging tools do you use?".  They blended together package and environment management (as does conda itself):
```

**Review by jaimergp — COMMENTED** [review]
Thanks Dave! I enjoyed the read a lot and could follow the presented facts easily. It's super informative to know what to work on next :D 

In addition to the comments, I noted some of the PSF images are a bit on the heavy side (600 KB). Those could use some extra compression / downscaling if that doesn't affect quality / readability too much.

That's all :) Thanks again for all this work!

File: blog/2023-03-10-conda-survey.md (hunk 7 above)
Comment (tnabtaf) [reply blog/2023-03-10-conda-survey.md:83]: Hmmm.  I don't think we need that period.  I think the question mark ends the sentence.

**Comment by tnabtaf** [comment]
Thanks @jaimergp .  I will get an updated commit in today.

File: blog/2023-03-10-conda-survey.md (hunk 1 above)
Comment (tnabtaf) [reply blog/2023-03-10-conda-survey.md]: Yes.  Will fix.

File: blog/2023-03-10-conda-survey.md (hunk 2 above)
Comment (tnabtaf) [reply blog/2023-03-10-conda-survey.md:130]: I have no idea what is happening here.  Can you recheck?

File: blog/2023-03-10-conda-survey.md (hunk 3 above)
Comment (tnabtaf) [reply blog/2023-03-10-conda-survey.md]: Went for all lower case.  I did not backtick anything as then I would have to decide when to also backtick conda.

File: blog/2023-03-10-conda-survey.md (hunk 5 above)
Comment (tnabtaf) [reply blog/2023-03-10-conda-survey.md:188]: Moved closing from the bottom of the top to the bottom of the bottom, and expanded it slightly

File: blog/2023-03-10-conda-survey.md (hunk 6 above)
Comment (tnabtaf) [reply blog/2023-03-10-conda-survey.md]: Bullet points are now consistent.

File: blog/2023-03-10-conda-survey.md (hunk 2 above)
Comment (jaimergp) [reply blog/2023-03-10-conda-survey.md:130]: Weird, it works now! But that's good!

**Comment by travishathaway** [comment]
Hi Dave,

Thanks for submitting this blog post! My constructive criticism is listed below:

### Content

Regarding the content, I believe we need to be much more explicit about what the survey represents and how it was conducted. I know you link to a Discourse post at the beginning, but I believe briefly mentioning how the survey was distributed and how responses were collected is important for any reader wishing to critically analyze this data.

In regards to representation, it's also important to be very explicit that this in no way represents the greater conda community because of the small sample size and the collection methods used (i.e. not randomized in any way and the inherit selection bias). In particular, I have a big problem with this quote:

> The conda survey had 72 responses. That's not huge, but it's not small either.

In my opinion, that wording is a little too ambiguous and it might mislead our readers into making more out of these numbers than they should. Instead, I believe it would better to use wording such as the following:

> While the survey itself only had 72 responses and is in no way representative of the entire conda community of users, the summary statistics listed here could point to particular problem areas that we may want to follow up on with further research or user interviews.

Small sample sizes are okay. Getting people to answer surveys is difficult. But, we need to acknowledge that and not let the data itself mislead readers. Especially considering some may not have a sufficient background in statistics to make that judgement.

### Accessibility

My first thought when seeing all the various charts and images was how accessible they would be for those trying reading the article with a screen reader. Unfortunately, it appears that a lot of the raw data is buried within the images. You do a good job of summarizing some of the data in the paragraphs though.

This isn't a blocker necessarily, but I did want to bring it up as I think it's important to be thinking about these types of things. We really do want this website to be as friendly to absolutely everyone as possible.

Possible solutions:
- Summarize the data in the "alt" text for the images (I can help with this)
- Include the raw as text at the bottom as an appendix for the article
- Include a table below the images with this information

This probably won't work for the PSF figures you included though as they are fairly complex and would include many numbers.

**Comment by tnabtaf** [comment]
Hi Travis, Thanks for the feedback.

> ### Content
> ... In particular, I have a big problem with this quote:
> 
> > The conda survey had 72 responses. That's not huge, but it's not small either.
> 
> In my opinion, that wording is a little too ambiguous and it might mislead our readers into making more out of these numbers than they should. Instead, I believe it would better to use wording such as the following:
> 
> > While the survey itself only had 72 responses and is in no way representative of the entire conda community of users, the summary statistics listed here could point to particular problem areas that we may want to follow up on with further research or user interviews.

I will make this change. I am hesitant to go beyond that statement though.  The post is already way long. 

> ### Accessibility
> My first thought when seeing all the various charts and images was how accessible they would be for those trying to read the article with a screen reader. Unfortunately, it appears that a lot of the raw data is buried within the images. You do a good job of summarizing some of the data in the paragraphs though.

Right now, all of the conda survey images point to backing spreadsheets in Google Drive.  I don't use screen readers, but I hope they would tell the reader that, and I would also hope that Google Sheets supports screen readers reasonably well.  Is that enough?

**Comment by travishathaway** [comment]
@tnabtaf,

Regarding accessibility concerns, I personally do not think this is enough. That still creates a very sub-par experience for these users in my opinion.

I'm not going to hold up this pull request because of this though. Instead, I'll bring it up in a future community meeting so we can properly discuss the importance of accessibility and to what extent we wish to support it for this website.

Thanks for reading and considering my feedback! ✌️ 😄

**Comment by travishathaway** [comment]
Hi @tnabtaf,

Once this pull request is merged:

- https://github.com/conda-incubator/conda-dot-org/pull/68

I am going to make some changes to this blog post based on it. Specifically, it has to do with the way that images are stored and referred to in the blog post itself.

**Comment by tnabtaf** [comment]
Travis,

> Regarding accessibility concerns, I personally do not think this is enough. That still creates a very sub-par experience for these users in my opinion.

Well, dang.  I'll do some research on best practices for presenting what is basically tabular data that is shared in an image.  I am all for getting this right.

> I am going to make some changes to this blog post based on it. Specifically, it has to do with the way that images are stored and referred to in the blog post itself.

I don't mind doing this either, once the infrastructure is enabled.  I actually favor distributing "one-time-use" graphics across the content tree (although that's just gut feel).

**Comment by travishathaway** [comment]
Hi @tnabtaf,

Thanks for digging in to the accessibility issue. I'll be interested to hear what you find out. In my experience, it really just means having as much information as possible available as text so that screen readers have access and making sure that we maintain the necessary contrasts with our colors to ensure readable text.

If you'd like to help with reorganizing the files for the pull request, I encourage you to look at the changes I recently made to a separate pull request:

- https://github.com/conda-incubator/conda-dot-org/pull/68

Before you do this though, we need to merge the above pull request, get the updates in this branch and then update your local install (if you are developing this locally).

Here's the highlights:

- For the blog post, create a folder called `blog/2023-03-10-conda-survey`; this is where the images and blog text go
- Create a file called `blog/2023-03-10-conda-survey/index.mdx`; this is where the text goes for the blog; please not the different file ending because it matters.
- Add all the images to a new folder called `blog/2023-03-10-conda-survey/img`

The last step is updating the blog post file (`blog/2023-03-10-conda-survey/index.mdx`). This will be the most tedious task 😭.

At the top of the file, just below the front matter, put this line:

```js
import Image from '@theme/IdealImage';
import surveyImageOne from './img/conda-is-my-preferred-package-manager.png';
// ... do this for all the images
```

Now, in the blog itself, replace the old references to the images with this:

```js
<Image img={surveyImageOne} alt="conda is my preferred package manager" />
```

Sorry that it's a bit complicated, but using this plugin does have several advantages (e.g. better page load times for a faster website).

Please let me know if you have any questions.

**Pull request merged**